sklearn/preprocessing/data.py

```diff
@@ -2866,6 +2869,229 @@ def power_transform(X, method='box-cox', standardize=True, copy=True):
     return pt.fit_transform(X)
 
 
+def _get_nested_counter(remaining, y_dim, inclusion_size):
```

Comment:
* _get_nested_counter: I know it's not a requirement to document private functions, but it would make the review
so much easier to know what the components are supposed to do. E.g., what is a 'nested dictionary', what do you
mean with layers and a 2d array of what in the end?

Reply:
There is something in the CI tests (something to do with the pickle module) that does not allow you to do something like  
a = {}
a[1] = {}
a[1][1] = 4  

So this is a workaround  

Should I add "This is a workaround due to some pickle issues in CI testing regarding creating nested dicts dynamically" to this as a comment?